File: src/backend/core/DTO/AuxDTO.cs
namespace SnippetVault {
using System;


public class TaskDTO {
    public int id {get; set;}
    public string name {get; set;}
    public string taskGroupName {get; set;}
    public string description {get; set;}
}

public class TaskGroupDTO {
    public int id {get; set;}
    public string name {get; set;}
    public string code {get; set;}
}

public class LanguageDTO {
    public int id {get; set;}
    public string name {get; set;}
    public string code {get; set;}
    public int sortingOrder { get; set; }
}

public class TaskCUDTO : CUDTO {
    public SelectChoice taskGroup {get; set;}
    public string name {get; set;}
    public string description {get; set;}
}

public class TaskGroupCUDTO : CUDTO {
    public string code {get; set;}
    public string name {get; set;}
}

public class LanguageCUDTO : CUDTO {
    public string name { get; set; }
    public string code { get; set; }
    public int sortingOrder { get; set; }
}

public class CUDTO {
    public int existingId {get; set;}
    public bool isDeleted {get; set;}
}

public class PostResponseDTO {
    public string status {get; set;}
}

}

File: src/backend/main/tech/sozonov/SnippetVault/core/DTO/AuxDTO.java
package tech.sozonov.SnippetVault.core.DTO;

import tech.sozonov.SnippetVault.core.utils.Types.SelectChoice;

public class AuxDTO {
public static class TaskDTO {
    public int id;
    public String name;
    public String taskGroupName;
    public String description;
}

public static class TaskGroupDTO {
    public int id;
    public String name;
    public String code;
}

public static class LanguageDTO {
    public int id;
    public String name;
    public String code;
    public int sortingOrder;
}

public static class TaskCUDTO extends CUDTO {
    public SelectChoice taskGroup;
    public String name;
    public String description;
}

public static class TaskGroupCUDTO extends CUDTO {
    public String code;
    public String name;
}

public static class LanguageCUDTO extends CUDTO {
    public String name;
    public String code;
    public int sortingOrder;
}

public static class CUDTO {
    public int existingId;
    public boolean isDeleted;
}

public static class PostResponseDTO {
    public String status;
}

}
